### PR TITLE
[core] Make task manager eventStream thread safe

### PR DESCRIPTION
**Pull request by teo**

### CODE REVIEW

File: core/environment/eventStream.go [hunk 1]
```diff
@@ -40,40 +39,45 @@ type Subscription interface {
...
+	defer e.mu.Unlock()
 	close(e.stream)
 	e.stream = nil
 }
 
 type eventSub struct {
-    feed         *eventStream
-    once         sync.Once
-    err          chan error
+	feed *eventStream
```
Comment (justonedev1) [inline core/environment/eventStream.go:71]: is there a reason for the change of format?

File: core/environment/environment.go [hunk 2]
```diff
@@ -987,7 +987,13 @@ func (env *Environment) scheduleAutoStopTransition() (scheduled bool, expected t
...
 							WithField("run", env.currentRunNumber).
 							Errorf("Scheduled auto stop transition failed: %s, Transitioning into ERROR", err.Error())
-						_ = env.TryTransition(NewGoErrorTransition(ManagerInstance().taskman))
+						err = env.TryTransition(NewGoErrorTransition(ManagerInstance().taskman))
+						if err != nil {
+							log.WithField("partition", env.id).
+								WithField("run", env.currentRunNumber).
+								Errorf("Forced transition to ERROR failed: %s", err.Error())
+							env.setState("ERROR")
+						}
```
Comment (justonedev1) [inline core/environment/environment.go:996]: does this part of code do something for the thread safety?

**Review by justonedev1 — COMMENTED** [review]
This seems ok if eventStream  is the only part that needs to be thread safe. However, I have few nitpicks, which I did not understand how those got there.

File: core/environment/environment.go (hunk 2 above)
Comment (teo) [reply core/environment/environment.go:996]: No, this bit ensures we go to ERROR even if the GO_ERROR transition fails for any reason.

**Review by justonedev1 — APPROVED** [review]

**Pull request merged**